src/ConfigurationStore.php

```diff
@@ -2,46 +2,85 @@
 
 namespace Eppo;
 
+use Eppo\Cache\CacheType;
+use Eppo\Cache\ICacheFactory;
+use Eppo\DTO\Flag;
+use Eppo\Exception\EppoClientException;
 use Psr\SimpleCache\CacheInterface;
 use Psr\SimpleCache\InvalidArgumentException;
 
-class ConfigurationStore
+class ConfigurationStore implements IConfigurationStore
 {
-    /** @var CacheInterface */
-    private $cache;
+    private CacheInterface $cache;
+    private CacheInterface $flagCache;
+
+    private CacheInterface $metadataCache;
+
+    const FLAG_TIMESTAMP = "flagTimestamp";
 
     /**
      * @param CacheInterface $cache
      */
     public function __construct(CacheInterface $cache)
     {
-        $this->cache = $cache;
+        $this->flagCache = new Cache\NamespaceCache(CacheType::FLAG, $cache);
+        $this->metadataCache = new Cache\NamespaceCache(CacheType::META, $cache);
     }
 
-    /**
-     * @param string $key
-     *
-     * @return array
-     *
-     * @throws InvalidArgumentException
-     */
-    public function getConfiguration(string $key): array
+    public function get(string $key): ?Flag
     {
-        $value = $this->cache->get($key);
-        return $value ? json_decode($value, true) : [];
+        try {
+            $result = $this->flagCache->get($key);
+            if ($result == null) return null;
+
+            $inflated = unserialize($result);
+            return $inflated === false ? null : $inflated;
+        } catch (InvalidArgumentException $e) {
+
+            // Simple cache throws exceptions when a keystring is not a legal value (characters {}()/@: are illegal)
+            syslog(LOG_WARNING, "[EPPO SDK] Illegal flag key ${key}: " . $e->getMessage());
+            return null;
+        }
+    }
+    private function setFlag(Flag $flag): void
+    {
+        try {
+            $this->flagCache->set($flag->key, serialize($flag));
+        } catch (InvalidArgumentException $e) {
+            $key = $flag->key;
+
+            // Simple cache throws exceptions when a keystring is not a legal value (characters {}()/@: are illegal)
+            syslog(LOG_WARNING, "[EPPO SDK] Illegal flag key ${key}: " . $e->getMessage());
+        }
     }
 
     /**
-     * @param array $configs
-     *
-     * @return void
-     *
-     * @throws InvalidArgumentException
+     * @throws EppoClientException
      */
-    public function setConfigurations(array $configs)
+    public function setFlags(array $flags): void
+    {
+        // Set last fetch timestamp.
+        try {
+            $this->metadataCache->set(self::FLAG_TIMESTAMP, time());
+        } catch (InvalidArgumentException $e) {
+            throw EppoClientException::From($e);
+        }
+
+        foreach($flags as $flag) {
```

Comment:
We'll want to clear the cache before doing this loop so that removed flags are purged